### PR TITLE
Test Cases for TraceStore

### PR DESCRIPTION
**Pull request by kasunben**
Test Cases:
- Server - Unit - TraceStore - registerSession
- Server - Unit - TraceStore - getTrace (Method)
- Server - Unit - TraceStore - getTrace (PubSub)
- Server - Unit - TraceStore - unregisterSession

### CODE REVIEW

File: tests/server/unit.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  traceStore._onMethodTrace(session, sampleTrace);
+
+  // get traces
+  var browserId = 'bid';
+  var clientId = 'cid';
+  var type = 'method';
+  var id = 'bidcid';
+
+  var trace = traceStore.getTrace(browserId, clientId, type, id);
+  test.isNotUndefined(trace);
```
Comment (arunoda) [inline tests/server/unit.js]: Do an equal check with the `sampleTrace`

File: tests/server/unit.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  traceStore._onSubTrace(session, sampleTrace);
+
+  // get trace
+  var browserId = 'bid';
+  var clientId = 'cid';
+  var type = 'pubsub';
+  var id = 'bidcid';
+
+  var trace = traceStore.getTrace(browserId, clientId, type, id);
+  test.isNotUndefined(trace);
```
Comment (arunoda) [inline tests/server/unit.js]: Do an equal check.

**Comment by arunoda** [comment]
I also need a following two tests
- Unit tests for `_buildTrace`
- Integration test for the method `kadira.debug.getTrace` (This is just to check the validity of the trace)

File: tests/server/integration.js [hunk 3]
```diff
@@ -81,6 +81,51 @@ function(test, done) {
...
+    // };
+
+    // receiver.traceStore._onSubTrace(sampleTrace, session);
+
+    // get trace
+    var type = 'pubsub';
+    var id = 'bidcid';
+
+    // Unless the method check is failed, this test will not fail.
+    var trace = sender.call('kadira.debug.getTrace', browserId, clientId, type, id);
```
Comment (arunoda) [inline tests/server/integration.js]: We need to do an equal check here. Do it and update the PR.
Then I can take it.

Comment (kasunben) [inline tests/server/integration.js]: Updated. Checked whether defined or not the trace object.

File: tests/server/integration.js [hunk 4]
```diff
@@ -81,8 +81,52 @@ function(test, done) {
...
+      userId: null
+    };
+
+    // get trace
+    var type = 'method';
+
+    sender.call('kadira.debug.getTrace', browserId, clientId, type, "0");
+    var trace = sender.call('kadira.debug.getTrace', browserId, clientId, type, "1");
+
+    test.isNotUndefined(trace);
```
Comment (arunoda) [inline tests/server/integration.js:95]: Check here for the trace's method Id.

Comment (arunoda) [inline tests/server/integration.js:95]: Remove additional spaces.

**Comment by kasunben** [comment]
Done.

**Comment by arunoda** [comment]
Great.

**Pull request merged**